### DISCUSSION TITLE
Visual Studio community: which install options fit Git-for-Windows basic code browsing, editing and compiling?

### DISCUSSION THREAD

**Discussion by PhilipOakley** [git-for-windows/git#4268] [Q&A]
The Visual Studio installer page is a mass of what feel to be irrelevant options for browsing the Git codebase, especially if full up builds are to be done via the SDK [Software Development Kit](https://github.com/git-for-windows/build-extra/releases/latest) or the CI (Continuous Integration) features. (many many Gb of 'stuff' in the install options)

I'm guessing that the "desktop development with C++" under 'Desktop & Mobile' might be the right one, though the "Universal Windows Platform development" maybe better.

While Usually I'm usually happy with a bit of Notepad++ to follow the code, having the VS intellisense can be a help in many instances. I feel I have some rubbish left in my older VS 2015 and 2019 setups, so don't want to repeat any previous mistakes.

It would be great to get some guidance on setting up Visual Studio so that more folks can 'follow along' with the code base and at least get a basic feel of Git.

What is our advice for those wanting . **to setup VS** . to look at the Git code.

Here's that happy page of options I'm being offered, though those blue ticks could be from my old installs :wink:

![VisualStudio selection Jan2023](https://user-images.githubusercontent.com/914343/216721168-0313270d-a851-4b58-aa69-eee608c0a4da.png)

**Comment by rimrul** [comment]
> I'm guessing that the "desktop development with C++" under 'Desktop & Mobile' might be the right one, 

I think that's the one.

> though the "Universal Windows Platform development" maybe better.

No, that one probably wont help you for most parts of git though that might be needed in addition to the previous one for WinToast.
